src/logger.cr

```diff
@@ -0,0 +1,72 @@
+class Logger
+  property :level, :progname, :formatter
+
+  UNKNOWN = 5
+  FATAL = 4
+  ERROR = 3
+  WARN = 2
+  INFO = 1
+  DEBUG = 0
+
+  SEV_LABEL = %w(DEBUG INFO WARN ERROR FATAL ANY)
+
+  DEFAULT_FORMATTER = ->(severity : String, datetime : Time, progname : String, message : String) {
+    "#{severity[0]}, [#{datetime} ##{Process.pid}] #{severity.rjust(5)} -- #{progname}: #{message}"
+  }
```

Comment:
For the above we could use a little but probably unknown feature. We can do this:

```crystal
# A function that formats log entries. It receives severity, datetime, progname and message.
alias Formatter = String, Time, String, String ->

DEFAULT_FORMATTER = Formatter.new do |severity, datetime, progname, message|
  "#{severity[0]}, [#{datetime} ##{Process.pid}] #{severity.rjust(5)} -- #{progname}: #{message}"
end
```

The good thing is that `Formatter` can be documented, and when you create one you don't need to specify the types over and over again. This is one of the few "magic" rules of Crystal: invoking `new` on a Function type with a block is automatically rewritten to a function literal with the correct types set. But instead of magic you can think every Function automatically defines such method.